src/GameConfiguration/index.ts

```diff
@@ -79,12 +106,74 @@ export default class GameConfiguration implements GameConfiguration {
   }
 
   update(visibleGameState: VisibleGameInformation): this {
+    const { grid, tiles } = this.revealed
+    const { passable, unknownObstacles, adjacencies, distances, cities, crowns } = this.hidden
+
+    function tileAt(rowIndex: number, colIndex: number): Tile | undefined {
+      const row = grid[rowIndex]
+      return row && row[colIndex]
+    }
+
+    function maybeAddAdjacency(tile: Tile, otherTile: Tile | undefined): void {
+      if (otherTile && passable.has(otherTile)) {
+        adjacencies.get(tile)!.add(otherTile)
+        adjacencies.get(otherTile)!.add(tile)
+        distances.get(tile)!.set(otherTile, 1)
+        distances.get(otherTile)!.set(tile, 1)
+      }
+    }
+
+    function markTileAsPassable(tile: Tile): void {
+      passable.add(tile)
+      distances.get(tile)!.set(tile, 0)
+      const { rowIndex, colIndex } = tile
+      const north = tileAt(rowIndex - 1, colIndex)
+      const south = tileAt(rowIndex + 1, colIndex)
+      const west  = tileAt(rowIndex, colIndex - 1)
+      const east  = tileAt(rowIndex, colIndex + 1)
+      
+      maybeAddAdjacency(tile, north)
+      maybeAddAdjacency(tile, south)
+      maybeAddAdjacency(tile, west)
+      maybeAddAdjacency(tile, east)
+
+      // update other distances which may now be shorter because of this new tile
+      for (const firstTile of passable) {
+        for (const secondTile of passable) {
+          const oldDistance = distances.get(firstTile)!.get(secondTile)
```

Comment:
Do we know for sure these distances will be numeric at this point? what if they're undefined?